### ISSUE TITLE
Some /api/quotas endpoints do not return json

### ISSUE THREAD

**Issue by innovate-invent** [galaxyproject/galaxy#11971]
`POST /api/quotas`
```
400 Bad Request
The server could not comply with the request since
it is either malformed or otherwise incorrect.
        
Quota names must be unique and a quota with that name already exists, so choose another name.
```

`POST /api/quotas/deleted/{encoded_quota_id}/undelete`
```
Undeleted 1 quotas: test
```

`PUT /api/quotas/{encoded_quota_id}`
```
Quota 'test' has been renamed to 'test'.; Quota 'test' is now '=1000 bytes'.; Quota 'test' is now the default for registered users.
```

**Comment by davelopez** [comment]
Yes, I noticed the raw string responses too when migrating this route in https://github.com/galaxyproject/galaxy/pull/11315

I assume we should maintain backward compatibility of the API in most cases but this particular case seems tied to the UI and can be easily fixed.

I am trying to identify these kinds of responses as I migrate endpoints to FastAPI. Hopefully, we can deprecate some of them in favor of more standard responses.

**Comment by innovate-invent** [comment]
I don't think there is any compatibility present with those responses, unless people are checking for substrings. That could be made backwards compatible by keeping the text response in a `{"message": ""}` field. A substring check will continue to work.
